docs/demo/scripts/customise_interactions.py

```diff
@@ -338,31 +338,14 @@ def custom_handler_function(x):
 
 
 # %%
-def wrap(func):
-    """
-
-    Args:
-        func:
-
-    Returns:
-
-    """
-
-    def pull_distributed_wrapper(x):
-        """
-
-        Args:
-            x:
-
-        Returns:
-
-        """
-        return func(x, tag="FWTW")
+from wsimod.extensions import extensions as extend
 
-    return pull_distributed_wrapper
 
+@extend.node_attribute(obj=my_fwtw, attribute_name="pull_distributed")
```

Comment:
I'm afraid things need to be a bit more elaborate than this since for this approach to work, you need to already have access to the formed objects. This is the case in this tutorial since you are creating all the components programmatically so you can pass the object, but it will not be true when including the extensions in an extension file. Indeed, in this case it is not necessary to use decorators as you can simply do: `my_fwtw.pull_distributed = new_distributed`, and that's all.

I suggest an alternative below.